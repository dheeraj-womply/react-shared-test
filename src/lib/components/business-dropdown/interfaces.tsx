export interface IBusinessDropdown {
    data: IBusinessDropdownData;
    handleChange: (args: any) => void;
}

export interface IBusinessDropdownData {
    businessDropdown: BusinessLocation[],
    activeLocation: BusinessLocation | undefined
}

export interface Address {
    address1: string;
    address2: string;
    city: string;
    state: string;
    zip: string;
}

export interface Category {
    id: number;
    name: string;
}

export interface Coordinates {
    latitude: number | null;
    longitude: number | null;
}

export interface Analytics {
    trial: boolean;
    is_trial: boolean;
    trial_end_date: string | null;
    claimed_at: string | null;
    claimed_at_month: string | null;
    sales_rep_id: string | null;
}

export interface EmailSettings {
    reputation_alerts: boolean;
    monthly_business_summaries: boolean;
    weekly_business_summaries: boolean;
    weekly_competitor_summaries: boolean;
}

export interface BusinessLocation {
    id: number;
    slug: string;
    name: string;
    business_source: string;
    business_phone: string;
    website_url: string;
    price: number | null;
    address: Address;
    category: Category;
    coordinates: Coordinates;
    analytics: Analytics;
    email_settings: EmailSettings;
}