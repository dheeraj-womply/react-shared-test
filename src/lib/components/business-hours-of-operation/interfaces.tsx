export interface IDay {
    day: string,
    hours : Array<IHoursRange>,
    isClosed: boolean
}

interface IHoursRange {
    openTime: string | null,
    closeTime: string | null
}

export interface ITimepickerProps {
    handleDateChange:(date: string | null, fieldName: 'openTime'|'closeTime', index:number) => void,
    fieldName: 'openTime'|'closeTime',
    value: string | null,
    index: number,
    disabled?: boolean
}

export interface IBusinessDaysOfWeek {
    day: string;
    openTime: string | null;
    closeTime: string | null;
}