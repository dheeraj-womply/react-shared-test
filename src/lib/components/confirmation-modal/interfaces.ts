export interface ConfirmationModalComponent {
    isModalOpen?: boolean;
    isMobile?: boolean;
    title: React.ReactNode;
    content:  React.ReactNode;
    primaryButtonText: string;
    onPrimaryButtonClick: () => any;
    secondaryButtonText: string;
    onSecondaryButtonClick: () => any;
}
