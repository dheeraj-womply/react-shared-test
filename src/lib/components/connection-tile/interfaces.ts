import { MouseEvent } from 'react'

export enum TileStatus {
    CONNECTED_VALID = 'CONNECTED_VALID',
    CONNECTED_INVALID = 'CONNECTED_INVALID',
    NOT_CONNECTED = 'NOT_CONNECTED',
    UNABLE_TO_CONNECT = 'UNABLE_TO_CONNECT',
    CONNECTION_PENDING = 'CONNECTION_PENDING',
}

export interface IConnectionItem {
    label: string;
    icon?: string;
    status: TileStatus;
}

export interface IConnectionTile {
    tile: IConnectionItem;
    onClick?: (event?: MouseEvent<HTMLDivElement>) => void;
}