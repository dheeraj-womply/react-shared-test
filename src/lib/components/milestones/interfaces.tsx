
export interface IMilestoneBubble {
    milestonesData: IMilestonesData;
    onClick: (item: IMilestone) => void;
}

export interface IMilestoneProgressCircle {
    milestonesData: IMilestonesData;
}

export interface IMilestonesData {
    milestones: IMilestone[];
    milestoneCount: number;
    completedMilestoneCount: number;
    pollingPages: IPollingPage[]
}

export interface IMilestone {
    key: string;
    description: string;
    displayText: string;
    urlPage: string;
    urlSubDomain: string;
    sortIndex: number;
    productId: number;
    completedTime: number;
}

export interface IPollingPage {
    milestoneKey: string;
    urlPage: string;
    urlSubDomain: string
}