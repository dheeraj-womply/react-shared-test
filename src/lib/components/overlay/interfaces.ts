export interface OverlayComponent {
  children?: React.ReactNode
  show?: boolean
  allowUserActions?: boolean
  hideSpinner?: boolean
  spinnerTop?: string
  rootStyles?: string
}
