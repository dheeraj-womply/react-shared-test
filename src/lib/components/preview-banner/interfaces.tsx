export interface IBannerProps {
    bannerId?: number;
    businessLocationId?: number;
    bannerAlignment?: string;
    displaySequence?: string|number;
    bannerType: string;
    isEnabled?: boolean;
    cssStyles: string; 
    startDate?: Date;
    endDate?: Date;
    bannerStart?: string;
    bannerFields: Array<IBannerFields>;
}

export interface IBannerFields {
    bannerFieldId: string;
    bannerFieldName: string;
    bannerFieldValue: string;
    bannerType: string;
    cssStyles: string;
    clickToAction?: string;
}