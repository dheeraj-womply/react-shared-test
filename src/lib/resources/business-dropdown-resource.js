export const businessDropdown = [
    {
    "id" : 3996325,
    "slug" : "authentic-plaster-and-til-ccpm",
    "name" : "AUTHENTIC PLASTER AND TIL",
    "address" : {
        "address1" : "1775 NORTHFORK LANE",
        "address2" : null,
        "city" : "LUCAS",
        "state" : "TX",
        "zip" : "75002-0000"
    },
    "category" : {
        "id" : 369,
        "name" : "Pools and Pool Supply Stores"
    },
    "coordinates" : {
        "latitude" : null,
        "longitude" : null
    },
    "price" : null,
    "analytics" : {
        "is_trial" : false,
        "trial_end_date" : null,
        "claimed_at" : null,
        "sales_rep_id" : null,
        "claimed_at_month" : null
    },
    "business_phone" : "9726701028",
    "website_url" : null,
    "email_settings" : {
        "monthly_business_summaries" : false,
        "reputation_alerts" : false,
        "weekly_business_summaries" : false,
        "weekly_competitor_summaries" : false
    },
    "business_source" : "CREDIT_CARD_PROCESSOR"
},
    {
    "id" : 3583577,
    "slug" : "orleans-auto-center-ccpm",
    "name" : "ORLEANS AUTO CENTER",
    "address" : {
        "address1" : "3841 1st St",
        "address2" : null,
        "city" : "ORLEANS",
        "state" : "MA",
        "zip" : "02653"
    },
    "category" : {
        "id" : 16,
        "name" : "Car Dealers"
    },
    "coordinates" : {
        "latitude" : null,
        "longitude" : null
    },
    "price" : null,
    "analytics" : {
        "is_trial" : false,
        "trial_end_date" : null,
        "claimed_at" : "2018-04-03",
        "sales_rep_id" : "ahuerta@womply.com",
        "claimed_at_month" : "04/2018"
    },
    "business_phone" : "+15082407978",
    "website_url" : "https://www.womply.com/",
    "email_settings" : {
        "monthly_business_summaries" : true,
        "reputation_alerts" : true,
        "weekly_business_summaries" : true,
        "weekly_competitor_summaries" : true
    },
    "business_source" : "CREDIT_CARD_PROCESSOR"
},
    {
    "id" : 3581390,
    "slug" : "seayama-market-llc-ccpm",
    "name" : "SEAYAMA MARKET LLC",
    "address" : {
        "address1" : "725 GEORGESVILLE RD",
        "address2" : null,
        "city" : "COLUMBUS",
        "state" : "OH",
        "zip" : "43228"
    },
    "category" : {
        "id" : 70,
        "name" : "Supermarkets and Grocery Stores"
    },
    "coordinates" : {
        "latitude" : null,
        "longitude" : null
    },
    "price" : null,
    "analytics" : {
        "is_trial" : false,
        "trial_end_date" : null,
        "claimed_at" : "2018-04-04",
        "sales_rep_id" : "alia@womply.com",
        "claimed_at_month" : "04/2018"
    },
    "business_phone" : "6144322601",
    "website_url" : null,
    "email_settings" : {
        "monthly_business_summaries" : true,
        "reputation_alerts" : false,
        "weekly_business_summaries" : false,
        "weekly_competitor_summaries" : false
    },
    "business_source" : "CREDIT_CARD_PROCESSOR"
}
];