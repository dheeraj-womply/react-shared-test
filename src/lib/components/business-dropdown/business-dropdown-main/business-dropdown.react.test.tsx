import React from 'react'
import renderer from 'react-test-renderer'
import { BusinessDropdown } from './index'
import { businessDropdown as businessDropdownItems } from '../../../resources/business-dropdown-resource'
import { render, fireEvent } from '@testing-library/react'
import Select from '@material-ui/core/Select'
import MenuItem from '@material-ui/core/MenuItem'
import { BusinessLocation } from '../interfaces'

const businessDropdown: BusinessLocation[]  = businessDropdownItems.slice() as any
const slug = 'authentic-plaster-and-til-ccpm'
const activeLocation = businessDropdown.find((item: BusinessLocation) => slug === item.slug) || businessDropdown[0]

test('Test snapshot for BusinessDropdown', () => {
    const component = renderer.create(
        <BusinessDropdown  handleChange={() => {}} data={{businessDropdown, activeLocation}}/>
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})

test('Test BusinessDropdown Select', async () => {
    const changeHandler = jest.fn()
    const {
        getByText,
        getByTestId,
        getByRole
    } = render(
        <Select
            value={activeLocation}
            data-testid='select'
            inputProps={{
                name: "age",
                id: "demo-controlled-open-select",
                "data-testid": "select-input"
            }}
            onChange={changeHandler}
        >
            {businessDropdown.map((item: any, index: number) => {
                return (<MenuItem key={index} value={item}>
                    {item.name}: {item.address.address1}
                    {item.address.city ? <span>&nbsp;{item.address.city} </span> : null}
                    {item.address.state ? <span>&nbsp;{item.address.state} </span> : null}
                    {item.address.zip ? <span>&nbsp;{item.address.zip} </span> : null}
                </MenuItem>)
            })}
        </Select>
    )

    const button = getByRole('button')
    expect(button).toHaveTextContent(/AUTHENTIC PLASTER AND TIL: 1775 NORTHFORK LANE LUCAS TX 75002-0000/i)

    const selectNode = getByTestId("select-input")
    const selectButton = getByRole("button")
    expect(selectButton).not.toBeNull()
    expect(selectNode).not.toBeNull()

    fireEvent.mouseDown(selectButton)
    const itemClickable = getByText(/ORLEANS AUTO CENTER/i)
    fireEvent.click(itemClickable)
    expect(changeHandler).toHaveBeenCalled()
})