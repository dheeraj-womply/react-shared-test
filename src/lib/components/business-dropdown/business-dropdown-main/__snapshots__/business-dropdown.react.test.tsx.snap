// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot for BusinessDropdown 1`] = `
<div
  className="sc-AxjAm AkoOq"
>
  <div
    className="MuiFormControl-root makeStyles-formControl-1"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span>
          AUTHENTIC PLASTER AND TIL: 1775 NORTHFORK LANE, LUCAS, TX 75002-0000
        </span>
      </div>
      <input
        aria-hidden={true}
        className="MuiSelect-nativeInput"
        onAnimationStart={[Function]}
        onChange={[Function]}
        required={false}
        tabIndex={-1}
        value={
          Object {
            "address": Object {
              "address1": "1775 NORTHFORK LANE",
              "address2": null,
              "city": "LUCAS",
              "state": "TX",
              "zip": "75002-0000",
            },
            "analytics": Object {
              "claimed_at": null,
              "claimed_at_month": null,
              "is_trial": false,
              "sales_rep_id": null,
              "trial_end_date": null,
            },
            "business_phone": "9726701028",
            "business_source": "CREDIT_CARD_PROCESSOR",
            "category": Object {
              "id": 369,
              "name": "Pools and Pool Supply Stores",
            },
            "coordinates": Object {
              "latitude": null,
              "longitude": null,
            },
            "email_settings": Object {
              "monthly_business_summaries": false,
              "reputation_alerts": false,
              "weekly_business_summaries": false,
              "weekly_competitor_summaries": false,
            },
            "id": 3996325,
            "name": "AUTHENTIC PLASTER AND TIL",
            "price": null,
            "slug": "authentic-plaster-and-til-ccpm",
            "website_url": null,
          }
        }
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
